### PR TITLE
[WIP] Kfconfig

### PR DESCRIPTION
**Pull request by gabrielwen**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/kubeflow/4271)

### CODE REVIEW

File: bootstrap/pkg/kfupgrade/kfupgrade.go [hunk 1]
```diff
@@ -74,7 +74,7 @@ func createNewKfApp(baseConfig string, oldKfDef *kfdefsv3.KfDef) (*kfdefsv3.KfDe
 	newKfDef.Spec.AppDir = newAppDir
 
 	// Make sure the new KfApp is created.
-	_, err = coordinator.CreateKfAppCfgFile(newKfDef)
+	_, err = coordinator.CreateKfAppCfgFileWithKfDef(newKfDef)
```
Comment (jlewi) [inline bootstrap/pkg/kfupgrade/kfupgrade.go]: Is this going to create merge conflicts?

**Comment by jlewi** [comment]
@gabrielwen and @lluunn How's this coming?
Can we start merging small PRs to minimize merge conflicts? E.g. can we pull out the changes to the config_converter and get those merged?

**Comment by lluunn** [comment]
@jlewi #4250 ready to merge

**Pull request closed without merging**